Islandora/islandora_paged_content
# 7.x islandora 1533

**Jira:** https://jira.duraspace.org/browse/ISLANDORA-1533
# What does this Pull Request do?

Adds functionality to optionally consolidate OCR from pages to their parent paged content objects during batch ingests, as well as to update paged content objects without consolidated OCR via Drush.
# How should this be tested?
- The Drush command 'paged-content-consolidate-missing-ocr' should be tested to make sure that it consolidates OCR datastreams to paged content objects accurately, and that it doesn't overwrite existing OCR datastreams.
- The 'islandora/object/PID/manage/book' and 'islandora/object/PID/manage/issue' management pages should be looked at; an 'Aggregate OCR to the parent?' checkbox has been added to the 'Perform OCR' section.
- Solr results altering options have been added to 'admin/islandora/solution_pack_config/paged_content'; these should be confirmed working.
- In all cases, it should be confirmed that these new features do not hijack settings on existing installs.
- There is a risk that, because we are generating OCR/HOCR both with the derivative generation hook _and_ with the batch ingest process, we may create unnecessary additional versions of OCR/HOCR datastreams. It should be confirmed that this isn't the case.
# Background context:

The purpose of ISLANDORA-1533 is to allow OCR datastreams to be aggregated and stored with the parent paged object so they can be indexed into Solr, and so that pages can be hidden from search results.
# Screenshots
## Paged Content Configuration (before)

![config before](https://cloud.githubusercontent.com/assets/3406327/10946386/65d72176-82f8-11e5-8a57-127c3b486f62.png)
## Paged Content Configuration (after)

![config after](https://cloud.githubusercontent.com/assets/3406327/10946384/65d33606-82f8-11e5-842e-93f33937a7a8.png)
## OCR Regeneration (before)

![ocr before](https://cloud.githubusercontent.com/assets/3406327/10946387/65d74c46-82f8-11e5-960c-35e36364e25f.png)
## OCR Regeneration (after)

![ocr after](https://cloud.githubusercontent.com/assets/3406327/10946385/65d6c424-82f8-11e5-8707-485859d70eb9.png)
# Additional Notes:
- **Does this change require documentation to be updated?** Possibly; most of the functionality will be described in the UI, but documentation may be necessary to inform of the overarching option here.
- **Does this change require any other modifications to be made to the repository (i.e., Regeneration activity, etc.)?** If the feature wants to be used, yes, the Drush script to consolidate OCR should be run.

This change depends on https://github.com/Islandora/islandora_ocr/pull/59

**Tagging:** @Islandora/7-x-1-x-committers @whikloj as component manager here.

---

QA Dan
_Developer_
**[discoverygarden inc.](http://www.discoverygarden.ca) | Managing Digital Content**
